File: android/src/main/java/top/huic/tencent_im_plugin/listener/CustomAdvancedMsgListener.java
package top.huic.tencent_im_plugin.listener;

import com.tencent.imsdk.v2.V2TIMAdvancedMsgListener;
import com.tencent.imsdk.v2.V2TIMMessage;
import com.tencent.imsdk.v2.V2TIMMessageReceipt;

import java.util.List;

import top.huic.tencent_im_plugin.TencentImPlugin;
import top.huic.tencent_im_plugin.entity.CustomMessageEntity;
import top.huic.tencent_im_plugin.enums.ListenerTypeEnum;

/**
 * 消息相关监听器
 */
public class CustomAdvancedMsgListener extends V2TIMAdvancedMsgListener {
    /**
     * 新消息通知
     */
    @Override
    public void onRecvNewMessage(V2TIMMessage msg) {
        super.onRecvNewMessage(msg);
        TencentImPlugin.invokeListener(ListenerTypeEnum.NewMessage, new CustomMessageEntity(msg));
    }

    /**
     * C2C已读回执
     */
    @Override
    public void onRecvC2CReadReceipt(List<V2TIMMessageReceipt> receiptList) {
        super.onRecvC2CReadReceipt(receiptList);
        TencentImPlugin.invokeListener(ListenerTypeEnum.C2CReadReceipt, receiptList);
    }

    /**
     * 消息撤回
     */
    @Override
    public void onRecvMessageRevoked(String msgID) {
        super.onRecvMessageRevoked(msgID);
        TencentImPlugin.invokeListener(ListenerTypeEnum.MessageRevoked, msgID);
    }
}

File: ios/Classes/listener/CustomAdvancedMsgListener.swift
//
// Created by 蒋具宏 on 2020/10/27.
//

import Foundation
import ImSDK

/// 消息相关监听器
class CustomAdvancedMsgListener: NSObject, V2TIMAdvancedMsgListener {
    /// 新消息通知
    func onRecvNewMessage(_ msg: V2TIMMessage!) {
        SwiftTencentImPlugin.invokeListener(type: ListenerType.NewMessage, params: MessageEntity.init(message: msg!))
    }

    /// C2C已读回执
    func onRecvC2CReadReceipt(_ receiptList: [V2TIMMessageReceipt]!) {
        var data: [[String: Any]] = [];
        for item in receiptList {
            data.append(CustomMessageReceiptEntity.getDict(info: item));
        }
        SwiftTencentImPlugin.invokeListener(type: ListenerType.C2CReadReceipt, params: data)
    }

    /// 消息撤回
    func onRecvMessageRevoked(_ msgID: String!) {
        SwiftTencentImPlugin.invokeListener(type: ListenerType.MessageRevoked, params: msgID)
    }
}
